File: TA5/tool-evaluation-4/HAMR/examples/test_data-port-periodic_domains_VM/both_vm/CAmkES_seL4_VM/src/c/ext-c/emitter_t_impl_Impl/emitter_t_impl_Impl.c
#include <emitter_t_impl_Impl_api.h>
#include <ext.h>

// This file will not be overwritten so is safe to edit

Unit CAmkES_seL4_VM_test_event_data_port_periodic_domains_emitter_t_impl_Impl_initialise_(
  STACK_FRAME
  CAmkES_seL4_VM_test_event_data_port_periodic_domains_emitter_t_impl_Impl this) {
 // example api usage

 api_logInfo__CAmkES_seL4_VM_test_event_data_port_periodic_domains_emitter_t_impl_Impl(this, string("Example logInfo"));
 api_logDebug__CAmkES_seL4_VM_test_event_data_port_periodic_domains_emitter_t_impl_Impl(this, string("Example logDebug"));
 api_logError__CAmkES_seL4_VM_test_event_data_port_periodic_domains_emitter_t_impl_Impl(this, string("Example logError"));
}

Unit CAmkES_seL4_VM_test_event_data_port_periodic_domains_emitter_t_impl_Impl_finalise_(
  STACK_FRAME
  CAmkES_seL4_VM_test_event_data_port_periodic_domains_emitter_t_impl_Impl this) {
  // example finalise method
}

Unit CAmkES_seL4_VM_test_event_data_port_periodic_domains_emitter_t_impl_Impl_timeTriggered_(
  STACK_FRAME
  CAmkES_seL4_VM_test_event_data_port_periodic_domains_emitter_t_impl_Impl this) {}

File: TA5/tool-evaluation-4/HAMR/examples/test_event_data_port_periodic_domains_VM/both_vm/CAmkES_seL4/src/main/component/both_vm/test_event_data_port_periodic_domains/emitter_t_impl_Impl.scala
// #Sireum

package both_vm.test_event_data_port_periodic_domains

import org.sireum._
import both_vm._

// This file will not be overwritten so is safe to edit
@record class emitter_t_impl_Impl (val api : emitter_t_impl_Bridge.Api) extends emitter_t_impl {

  override def initialise(): Unit = {
    // example api usage

    api.logInfo("Example info logging")
    api.logDebug("Example debug logging")
    api.logError("Example error logging")

    api.sendwrite_port(Base_Types.Integer_8_empty())
  }

  override def timeTriggered(): Unit = {
    // example override of timeTriggered
  }

  override def activate(): Unit = {
    // example override of activate
  }

  override def deactivate(): Unit = {
    // example override of deactivate
  }

  override def finalise(): Unit = {
    // example override of finalise
  }

  override def recover(): Unit = {
    // example override of recover
  }
}
